server/sttp-stub-server/src/test/scala/sttp/tapir/server/stub/TapirStubInterpreterTest.scala

```diff
@@ -204,8 +205,90 @@ class TapirStubInterpreterTest extends AnyFlatSpec with Matchers {
     response.body shouldBe Left("Internal server error")
     response.code shouldBe StatusCode.InternalServerError
   }
+
+  it should "handle multipart body" in {
+    // given
+    val e =
+      endpoint.post
+        .in("api" / "multipart")
+        .in(multipartBody)
+        .out(stringBody)
+
+    val server = TapirStubInterpreter(SttpBackendStub(IdMonad))
+      .whenEndpoint(e)
+      .thenRespond("success")
+      .backend()
+
+    // when
+    val response = sttp.client3.basicRequest
+      .post(uri"http://test.com/api/multipart")
+      .multipartBody(multipart("name", "abc"))
+      .send(server)
+
+    // then
+    response.body shouldBe Right("success")
+  }
+
+  it should "correctly process a multipart body" in {
```

Comment:
these tests look much better now, can you maybe extend these two cases to handle multipart parts, also including files (as that's the most common use-case for multiparts)?